routes/api/project/location/private_subnet.rb

```diff
@@ -37,8 +37,8 @@ class CloverApi
           if request_body_params["firewall_id"]
             firewall_id = request_body_params["firewall_id"]
             fw = Firewall.from_ubid(firewall_id)
-            unless fw
-              fail Validation::ValidationFailed.new(firewall_id: "Firewall with id \"#{firewall_id}\" not found")
+            unless fw && fw.location == @location
+              fail Validation::ValidationFailed.new(firewall_id: "Firewall with id \"#{firewall_id}\" and location \"#{@location}\" is not found")
```

Comment:
```suggestion
              fail Validation::ValidationFailed.new(firewall_id: "Firewall with id \"#{firewall_id}\" is not found in location \"#{@location}\"")
```

Reply:
Since the suggestion implies that the firewall with that id exits but not in the given location, I'm keeping it as it is.